pkg/scaling/scale_handler.go

```diff
@@ -576,81 +587,190 @@ func (h *scaleHandler) getScaledObjectState(ctx context.Context, scaledObject *k
 	// Let's collect status of all scalers, no matter if any scaler raises error or is active
 	scalers, scalerConfigs := cache.GetScalers()
 	for scalerIndex := 0; scalerIndex < len(scalers); scalerIndex++ {
-		scalerName := strings.Replace(fmt.Sprintf("%T", scalers[scalerIndex]), "*scalers.", "", 1)
-		if scalerConfigs[scalerIndex].TriggerName != "" {
-			scalerName = scalerConfigs[scalerIndex].TriggerName
+		req := getStateScalerRequest{
+			ctx:          ctx,
+			scaler:       scalers[scalerIndex],
+			scalerIndex:  scalerIndex,
+			scalerConfig: scalerConfigs[scalerIndex],
+			cache:        cache,
+			logger:       logger,
+			scaledObject: scaledObject,
 		}
 
-		metricSpecs, err := cache.GetMetricSpecForScalingForScaler(ctx, scalerIndex)
+		result :=
+			h.getScalerState(req)
```

Comment:
```suggestion
		result := h.getScalerState(req)
```

Reply:
I would prefer if we don't use the dto here

Reply:
removed